lib/ui/popup.js

```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const EXPORTED_SYMBOLS = ['Popup'];
+
+function Popup(opts) {
+  this.win = opts.win;
+  this.events = opts.events;
+
+  this.onPopupHiding = this.onPopupHiding.bind(this);
```

Comment:
Yeah, this is a weird one. In this case the name of the event is actually `popuphiding`, hence, `onPopupHiding`: https://developer.mozilla.org/en-US/docs/Web/Events/popuphiding.

The event is sent just before the popup hides itself, so maybe `beforePopupHide` or `beforePopupClose` would be more expressive here?